### PR TITLE
Check for the case where the Operators vector in a tflite model is missing

### PR DESCRIPTION
**Pull request by shlmregev**
Some of our tools remove the vector if it's empty (i.e. the subgraph doesn't contain ops). The check is added until the tools are fixed.

Note that to properly build the code, we needed to add `#define FLATBUFFERS_LOCALE_INDEPENDENT 0` to a number of sources and headers in the TFLM code. http://b/193264978 has more context on why that was needed and what options we might have to get the same outcome in a more maintainable manner.

BUG=http://b/192589496

### CODE REVIEW

**Review by njeffrie — APPROVED** [review]
Looks good. Hopefully we can revert this once the flatbuffer API respects empty subgraphs. Perhaps reference a bug indicating this in the PR description.

File: tensorflow/lite/micro/micro_graph.cc [hunk 1]
```diff
@@ -94,7 +101,14 @@ TfLiteStatus MicroGraph::PrepareSubgraphs() {
     current_subgraph_index_ = subgraph_idx;
 
     const SubGraph* subgraph = (*subgraphs_)[subgraph_idx];
-    for (size_t i = 0; i < subgraph->operators()->size(); ++i) {
```
Comment (njeffrie) [inline tensorflow/lite/micro/micro_graph.cc:97]: Do you think we can split this repeated code into a helper method?

File: tensorflow/lite/micro/micro_interpreter.cc [hunk 2]
```diff
@@ -96,7 +96,15 @@ TfLiteStatus MicroInterpreter::PrepareNodeAndRegistrationDataFromFlatbuffer() {
     auto* opcodes = model_->operator_codes();
     BuiltinDataAllocator* builtin_data_allocator =
         allocator_.GetBuiltinDataAllocator();
-    for (size_t i = 0; i < subgraph->operators()->size(); ++i) {
+
+    // TODO(b/192589496): Ops must always be there. Remove this check when fixed
+    int operators_size;
```
Comment (njeffrie) [inline tensorflow/lite/micro/micro_interpreter.cc:101]: Maybe even this can call the helper method, if it's a public method in MicroGraph.

**Review by njeffrie — CHANGES_REQUESTED** [review]

File: tensorflow/lite/micro/micro_graph.cc (hunk 1 above)
Comment (shlmregev) [reply tensorflow/lite/micro/micro_graph.cc:97]: Done

**Review by njeffrie — DISMISSED** [review]

File: tensorflow/lite/micro/micro_graph.cc [hunk 3]
```diff
@@ -244,4 +241,14 @@ TfLiteEvalTensor* MicroGraph::GetSubgraphOutput(int subgraph_idx,
   return &subgraph_allocations_[subgraph_idx].tensors[tensor_idx];
 }
 
+int MicroGraph::NumSubgraphOperators(int subgraph_idx) {
```
Comment (advaitjain) [inline tensorflow/lite/micro/micro_graph.cc:244]: there isn't any reason for this function to be a public member of the MicroGraph class.

Can we make it a free function:
```cc
int NumSubgraphOperators(const Subgraph& subgraph);
```

And have this function be used across micro_graph.cc and micro_allocator.cc?

I'd be ok with putting this and the code from https://github.com/tensorflow/tflite-micro/pull/243 into a flatbuffer_utils.h/flatbuffer_utils.cc module.

**Review by advaitjain — CHANGES_REQUESTED** [review]

File: tensorflow/lite/micro/micro_graph.cc (hunk 3 above)
Comment (shlmregev) [reply tensorflow/lite/micro/micro_graph.cc:244]: Is this the path you had in mind:
tensorflow/lite/micro/flatbuffer_utils.*

Comment (advaitjain) [reply tensorflow/lite/micro/micro_graph.cc:244]: yup

File: tensorflow/lite/micro/micro_graph.h [hunk 4]
```diff
@@ -60,12 +60,15 @@ class MicroGraph {
...
 
-  // Number of tensor outputs to a specified subgraph in the model.
+  // Number of tensor outputs from a specified subgraph in the model.
   virtual size_t NumSubgraphOutputs(int subgraph_idx);
 
   // Get the specified output tensor of a specified subgraph in the model.
   virtual TfLiteEvalTensor* GetSubgraphOutput(int subgraph_idx, int output_idx);
 
+  // Number of operators in the specified subgraph in the model.
+  virtual int NumSubgraphOperators(int subgraph_idx);
```
Comment (advaitjain) [inline tensorflow/lite/micro/micro_graph.h:70]: I think I got confused in the internal review of this change. If this is a new function added in the current change, I think it is not needed anymore (now that we have the free functions in flatbuffer_utils.cc).

File: tensorflow/lite/micro/micro_graph.cc (hunk 3 above)
Comment (advaitjain) [reply tensorflow/lite/micro/micro_graph.cc:245]: I think this function is now unused and can be removed?

**Review by advaitjain — CHANGES_REQUESTED** [review]
One clarification about whether we still need `MicroGraph::NumSubgraphOperators`.

Other than that, lgtm.

File: tensorflow/lite/micro/micro_graph.h (hunk 4 above)
Comment (shlmregev) [reply tensorflow/lite/micro/micro_graph.h:70]: I made a mess by merging the origin branch before pushing my update. It overwrote my changes.

Comment (shlmregev) [reply tensorflow/lite/micro/micro_graph.h]: OK, I think I fixed it.

**Review by advaitjain — DISMISSED** [review]

**Review by advaitjain — APPROVED** [review]

**Review by advaitjain — APPROVED** [review]

**Review by advaitjain — DISMISSED** [review]

**Review by advaitjain — APPROVED** [review]

**Pull request merged**